### PR TITLE
Read and write consistency level options

### PR DESCRIPTION
**Pull request by fvcaputo**
## Why? :open_book:
So we can set read and write consistency levels on Cassandra, which can have impacts on Clusters on multiple regions.

## What? :wrench:
Add theses values as possible options, but [keep the default values if they are not set](https://github.com/datastax/spark-cassandra-connector/blob/master/doc/reference.md).

Also change Cassandra Client with these options, they are the same as the previous one but they will take into consideration multiple regions if they are available.

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## How everything was tested? :straight_ruler:
Mostly enable configuration changes and all tests should still be working.

## Checklist
- [x] My code follows the style guidelines of this project (docstrings, type hinting and linter compliance);
- [x] I have performed a self-review of my own code;
- [x] I have made corresponding changes to the documentation;
- [x] I have added tests that prove my fix is effective or that my feature works;
- [x] New and existing unit tests pass locally with my changes;
- [x] Add labels to distinguish the type of pull request. Available labels are `bug`, `enhancement`, `feature`, and `review`.

### CODE REVIEW

File: butterfree/configs/db/cassandra_config.py [hunk 1]
```diff
@@ -150,6 +156,28 @@ def stream_checkpoint_path(self, value: str) -> None:
...
+    @property
+    def read_consistency_level(self) -> Optional[str]:
+        """Read consistency level for Spark."""
+        return self.__read_consistency_level
+
+    @read_consistency_level.setter
+    def read_consistency_level(self, value: str) -> None:
+        self.__read_consistency_level = value or environment.get_variable(
+            "CASSANDRA_READ_CONSISTENCY_LEVEL", "LOCAL_ONE"
+        )
```
Comment (allrod5) [inline butterfree/configs/db/cassandra_config.py:168]: It is a bit odd to image a call with `value=None` and have this function to try to read from an env var. I would expect a regular assignment, at most an error/warning if trying to set an invalid value such as `None` (is it invalid?).

**Review by allrod5 — DISMISSED** [review]
I've just some small considerations but this is good to go as I see it though it would be best to wait for a reviewer from the @quintoandar/mlcops team too.

File: butterfree/configs/db/cassandra_config.py [hunk 2]
```diff
@@ -150,6 +154,24 @@ def stream_checkpoint_path(self, value: str) -> None:
             "STREAM_CHECKPOINT_PATH"
         )
 
+    @property
+    def read_consistency_level(self) -> Optional[str]:
+        """Read consistency level for Spark."""
```
[the remark this answers is not in the document]
Comment (fvcaputo) [reply butterfree/configs/db/cassandra_config.py:161]: Ah you are right.

File: butterfree/configs/db/cassandra_config.py (hunk 1 above)
Comment (fvcaputo) [reply butterfree/configs/db/cassandra_config.py:168]: I sorta followed already existing setters like 
https://github.com/quintoandar/butterfree/blob/b4b7628a32e172cf5d631d2d89674f45848bdd59/butterfree/configs/db/cassandra_config.py#L155
Where if value = None (nothing is passed) then it defaults to the env var. The difference in this one is that even if no env var was set, I kept the sprk default

Comment (moromimay) [reply butterfree/configs/db/cassandra_config.py:168]: That's right @fvcaputo :wink:

File: tests/unit/butterfree/configs/db/test_cassandra_config.py [hunk 3]
```diff
@@ -159,6 +159,50 @@ def test_stream_checkpoint_path_custom(self, cassandra_config):
...
+        value = "Custom Config"
+        cassandra_config.write_consistency_level = value
+
+        # then
+        assert cassandra_config.write_consistency_level == value
+
+    def test_write_consistency_level_custom_env_var(self, mocker, cassandra_config):
+        # given
+        value = "Custom Config"
+        mocker.patch("butterfree.configs.environment.get_variable", return_value=value)
```
Comment (moromimay) [inline tests/unit/butterfree/configs/db/test_cassandra_config.py:200]: Why are you using mocker? :thinking:

File: tests/unit/butterfree/configs/db/test_cassandra_config.py [hunk 4]
```diff
@@ -159,6 +159,50 @@ def test_stream_checkpoint_path_custom(self, cassandra_config):
...
+        value = "Custom Config"
+        cassandra_config.read_consistency_level = value
+
+        # then
+        assert cassandra_config.read_consistency_level == value
+
+    def test_read_consistency_level_custom_env_var(self, mocker, cassandra_config):
+        # given
+        value = "Custom Config"
+        mocker.patch("butterfree.configs.environment.get_variable", return_value=value)
```
Comment (moromimay) [inline tests/unit/butterfree/configs/db/test_cassandra_config.py:178]: ditto

**Review by moromimay — APPROVED** [review]
:star2:

**Pull request merged**